### DISCUSSION TITLE
How to generate JSON & YAML from org.eclipse.microprofile.openapi.models.OpenAPI instance

### DISCUSSION THREAD

**Discussion by dhoffer** [eclipse/microprofile-open-api#668] [Q&A]
I have a use case where I have to modify the org.eclipse.microprofile.openapi.models.OpenAPI model per user's JAX-RS request.  I can do the filtering of that model but now how do I save the modified model to JSON & YAML?

**Comment by Azquelt** [comment]
This is not possible with the current spec. The OpenAPI model is created at startup and cannot be changed later.

You're welcome to open an enhancement request issue outlining your use case and optionally proposing how it could be implemented.

**Comment by dhoffer** [comment]
@Azquelt I'm not entirely clear regarding your answer.  The OpenAPI model is created on the first request to generate it but then we use an OASFilter to heavily modify it.  So its not actually generated at startup but on first request.   But there is a quarkus property smallrye-openapi.always-run-filter: true that says it will run the OASFilter on each request.  So it seems it should be possible to run the OASFilter on each reach request or am I misunderstanding what the  smallrye-openapi.always-run-filter does?  I'm going to test it out in a minute.

I tested the quarkus.smallrye-openapi.always-run-filter = true property and it does nothing.  Filter runs on first request only.  I'll file a bug report.

**Comment by dhoffer** [comment]
@Azquelt I did finally get quarkus.smallrye-openapi.always-run-filter = true property to work so its called on each request, so now the question is how to I pass per run data into the entry point of the process which is /q/swagger-openapi?  I just need to pass a user's name so could that be a query parameter?  What is the purpose of a always run filter w/o a way to pass per run data?

**Reply by phillip-kruger** [reply]
@dhoffer - It would be better to discuss this on the Quarkus issue. It's really a Quarkus feature.
